Removed limitation for --cookie-jar to use only one hurl file

lilyhuang-github opened the pull request:
This removes the limitation of only allowing for one hurl file with the --cookie-jar argument and updates the documentation as such.
Should Closes #2537

jcamiel commented on docs/standalone/hurl-5.0.1.html:
```diff
@@ -2971,7 +2971,7 @@ <h3 id="getting-started-manual-options"><a href="#getting-started-manual-options
     <tr>
       <td><a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a></td>
-      <td>Write cookies to FILE after running the session (only for one session).<br />The file will be written using the Netscape cookie file format.<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
+      <td>Write cookies to FILE after running the session.<br />The file will be written using the Netscape cookie file format.<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
```
This file musn't be changed as it's the doc for v5.0.1

jcamiel commented on docs/standalone/hurl-5.0.1.md:
```diff
@@ -785,7 +785,7 @@ will follow a redirection only for the second entry.
 | <a href="#getting-started-manual-continue-on-error" id="getting-started-manual-continue-on-error"><code>--continue-on-error</code></a>                          | Continue executing requests to the end of the Hurl file even when an assert error occurs.<br>By default, Hurl exits after an assert error in the HTTP response.<br><br>Note that this option does not affect the behavior with multiple inp ...
 | <a href="#getting-started-manual-cookie" id="getting-started-manual-cookie"><code>-b, --cookie &lt;FILE&gt;</code></a>                                          | Read cookies from FILE (using the Netscape cookie file format).<br><br>Combined with [`-c, --cookie-jar`](#getting-started-manual-cookie-jar), you can simulate a cookie storage between successive Hurl runs.<br><br>This is a cli-only op ...
-| <a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a>                              | Write cookies to FILE after running the session (only for one session).<br>The file will be written using the Netscape cookie file format.<br><br>Combined with [`-b, --cookie`](#getting-started-manual-cookie), you can simulate a cookie ...
+| <a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a>                              | Write cookies to FILE after running the session.<br>The file will be written using the Netscape cookie file format.<br><br>Combined with [`-b, --cookie`](#getting-started-manual-cookie), you can simulate a cookie storage between succes ...
```
This file musn't be changed as it's the doc for v5.0.1

jcamiel commented on docs/standalone/hurl-6.0.0.html:
```diff
@@ -2973,7 +2973,7 @@ <h3 id="getting-started-manual-options"><a href="#getting-started-manual-options
     <tr>
       <td><a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a></td>
-      <td>Write cookies to FILE after running the session (only for one session).<br />The file will be written using the Netscape cookie file format.<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
+      <td>Write cookies to FILE after running the session.<br />The file will be written using the Netscape cookie file format.<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
```
This file musn't be changed as it's the doc for v6.0.0

jcamiel commented on docs/standalone/hurl-6.0.0.md:
```diff
@@ -787,7 +787,7 @@ will follow a redirection only for the second entry.
 | <a href="#getting-started-manual-continue-on-error" id="getting-started-manual-continue-on-error"><code>--continue-on-error</code></a>                          | Continue executing requests to the end of the Hurl file even when an assert error occurs.<br>By default, Hurl exits after an assert error in the HTTP response.<br><br>Note that this option does not affect the behavior with multiple inp ...
 | <a href="#getting-started-manual-cookie" id="getting-started-manual-cookie"><code>-b, --cookie &lt;FILE&gt;</code></a>                                          | Read cookies from FILE (using the Netscape cookie file format).<br><br>Combined with [`-c, --cookie-jar`](#getting-started-manual-cookie-jar), you can simulate a cookie storage between successive Hurl runs.<br><br>This is a cli-only op ...
-| <a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a>                              | Write cookies to FILE after running the session (only for one session).<br>The file will be written using the Netscape cookie file format.<br><br>Combined with [`-b, --cookie`](#getting-started-manual-cookie), you can simulate a cookie ...
+| <a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a>                              | Write cookies to FILE after running the session.<br>The file will be written using the Netscape cookie file format.<br><br>Combined with [`-b, --cookie`](#getting-started-manual-cookie), you can simulate a cookie storage between succes ...
```
This file musn't be changed as it's the doc for v6.0.0

jcamiel commented on packages/hurl/src/main.rs:
```diff
@@ -305,17 +305,13 @@ fn create_cookies_file(
+    }
+    for run in runs.iter() {
+        for cookie in run.hurl_result.cookies.iter() {
+            s.push_str(&cookie.redact(secrets));
```
As the Netscape file format for cookie support comment (`# some comment`) maybe we can add a preliminary  comment on the file used something like `# Cookies for file <tests_ok/test.hurl>`. This will help separating the different cookies.

jcamiel reviewed (changes requested):
Hi @lilyhuang-github some changes to be address; also could you please squash you commit to a single one and you have your commits to be signed (tehre is so information on it in the [CONTRIBUTING guide](https://github.com/Orange-OpenSource/hurl/blob/master/CONTRIBUTING.md)).

<img width="1448" alt="sign" src="https://github.com/user-attachments/assets/9aebd6ed-8be2-4995-b5dd-e140d1540d2b" />

lilyhuang-github commented:
> Hi @lilyhuang-github some changes to be address; also could you please squash you commits to a single one and you have your commits to be signed (there is so information on it in the [CONTRIBUTING guide](https://github.com/Orange-OpenSource/hurl/blob/master/CONTRIBUTING.md)).
> <img alt="sign" width="1448" src="https://private-user-images.githubusercontent.com/16323814/411303671-9aebd6ed-8be2-4995-b5dd-e140d1540d2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkxNTg1NjgsIm5iZiI6MTczOTE1ODI2OCwicGF0aCI6Ii8xNjMyMzgxNC80MTEzMDM2NzEtOWFlYmQ2ZWQtOGJlMi00OTk1LWI1ZGQtZTE0MGQxNTQwZDJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEwVDAzMzEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjMmE3ZDVlNDFjYWRjYjQ2MDEwY2UxZWJjMzc0NzFhNGVhYTUzZjY4ZjczMjM5YWI3MGRlZTc0NjI0NjkwZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.I7lIYmV4MsCTRfZXxAVmSHJJMNtmuDJuHoIYX1agZWQ">

Hi, I've implemented the changes, squashed into one commit and it should be verified/signed now. Sorry about accidentally changing the old documentation.

jcamiel commented on docs/standalone/hurl-5.0.1.html:
```diff
@@ -2971,7 +2971,7 @@ <h3 id="getting-started-manual-options"><a href="#getting-started-manual-options
     <tr>
       <td><a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a></td>
-      <td>Write cookies to FILE after running the session (only for one session).<br />The file will be written using the Netscape cookie file format.<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
+      <td>Write cookies to FILE after running the session.<br />The file will be written using the Netscape cookie file format (only for one session).<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
```
This file musn't be changed

jcamiel commented on docs/standalone/hurl-6.0.0.html:
```diff
@@ -2973,7 +2973,7 @@ <h3 id="getting-started-manual-options"><a href="#getting-started-manual-options
     <tr>
       <td><a href="#getting-started-manual-cookie-jar" id="getting-started-manual-cookie-jar"><code>-c, --cookie-jar &lt;FILE&gt;</code></a></td>
-      <td>Write cookies to FILE after running the session (only for one session).<br />The file will be written using the Netscape cookie file format.<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
+      <td>Write cookies to FILE after running the session.<br />The file will be written using the Netscape cookie file format (only for one session).<br /><br />Combined with <a href="#getting-started-manual-cookie"><code>-b, --cookie</code></a>, you can simulate a cookie storage between successive Hurl runs.<br /><br />This is a cli-only option.<br /></td>
```
This file musn't be changed

jcamiel reviewed (changes requested):
Hi @lilyhuang-github there are still modifications on files that musn't be changed

lilyhuang-github commented:
> Hi @lilyhuang-github there are still modifications on files that musn't be changed

Sorry, I thought I changed it back. It should be correct now.

lilyhuang-github commented:
I accidentally added the (only  for one session) twice but it should be correct now.

lilyhuang-github commented:
I changed the integration test output for the cookie-jar since it changed. If it's ok I can rebase it. If it's not I can undo the commit.

jcamiel commented:
Hi @lilyhuang-github it's perfect!

jcamiel commented:
Hi @lilyhuang-github you have to change also `integration/hurl/tests_ok/help.out.pattern` 

```
-c, --cookie-jar <FILE>  Write cookies to FILE after running the session (only for one session)
```

to

```
-c, --cookie-jar <FILE>  Write cookies to FILE after running the session
```

lilyhuang-github commented:
> Hi @lilyhuang-github you have to change also `integration/hurl/tests_ok/help.out.pattern`
> 
> ```
> -c, --cookie-jar <FILE>  Write cookies to FILE after running the session (only for one session)
> ```
> 
> to
> 
> ```
> -c, --cookie-jar <FILE>  Write cookies to FILE after running the session
> ```

I've added this change and also changed the README.md's cookie-jar explaination and rebased.

jcamiel commented:
@lilyhuang-github just a small fix:

```
error: `to_string` applied to a type that implements `Display` in `format!` args
   --> packages/hurl/src/main.rs:314:25
    |
314 |             run.filename.to_string()
    |                         ^^^^^^^^^^^^ help: remove this
    |
```

lilyhuang-github commented:
> @lilyhuang-github just a small fix:
> 
> ```
> error: `to_string` applied to a type that implements `Display` in `format!` args
>    --> packages/hurl/src/main.rs:314:25
>     |
> 314 |             run.filename.to_string()
>     |                         ^^^^^^^^^^^^ help: remove this
>     |
> ```

I've added that fix and rebased now.

jcamiel commented:
Hi,
You must run `cargo fmt` on your code change

lilyhuang-github commented:
> Hi, You must run `cargo fmt` on your code change

Sorry, I've ran the formatting with it now.

jcamiel commented:
@lilyhuang-github you've to run `cargo fmt` at the base of your repo, squash the changes and repush in order to appease the CI/CD

lilyhuang-github commented:
> @lilyhuang-github you've to run `cargo fmt` at the base of your repo, squash the changes and repush in order to appease the CI/CD

Sorry, I've done that now.

jcamiel commented:
/accept
